flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -259,9 +260,10 @@ BufferAndBacklog pollBuffer() {
 			}
 
 			while (!buffers.isEmpty()) {
-				BufferConsumer bufferConsumer = buffers.peek().getBufferConsumer();
+				BufferConsumerWithPartialRecordLength bufferConsumerWithPartialRecordLength = buffers.peek();
+				BufferConsumer bufferConsumer = requireNonNull(bufferConsumerWithPartialRecordLength).getBufferConsumer();
```

Comment:
I think there is no point in adding explicit `requireNonNull` just before dereferencing it.

Reply:
add here to avoid IDE check warning.

Reply:
I think this IDE warning doesn't make sense here so.
I'd prefer to alter IDE behavior rather than code (e.g. by adding `@SuppressWarnings("ConstantConditions")`).

Reply:
I agree, let's just remove it.

Reply:
Thanks!